source/FreeRTOS_DHCP.c

```diff
@@ -1455,7 +1447,7 @@
                  * it easier to lookup a device in a router's list of DHCP clients. */
 
                 /* Point to where the OPTION_END was stored to add data. */
-                pucPtr = &( pucUDPPayloadBuffer[ dhcpFIRST_OPTION_BYTE_OFFSET + ( *pxOptionsArraySize - 1U ) ] );
+                uint8_t * pucPtr = &( pucUDPPayloadBuffer[ dhcpFIRST_OPTION_BYTE_OFFSET + ( *pxOptionsArraySize - 1U ) ] );
```

Comment:
uint8_t * pucPtr should be declared at line 1400, any reason for moving it here?

Reply:
`pucPtr` is only used when `#if ( ipconfigDHCP_REGISTER_HOSTNAME == 1 )` evaluates to true.